lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb

```diff
@@ -0,0 +1,139 @@
+require_relative 'miq_ec2_vmbase'
+require_relative '../instance_metadata'
+
+module AmazonSsaSupport
+  class MiqEC2EbsVmBase < MiqEC2VmBase
+    attr_reader :volumes
+
+    def initialize(ec2_obj, host_instance, ec2, iargs)
+      super
+      @block_device_keys = ebs_ids
+      @volumes = []
+      @miqVm = nil
+    end
+  
+    def ebs_instance?
+      @ec2_obj.respond_to?(:instance_id) ? true : false
+    end
+    
+    def ebs_image?
+      @ec2_obj.respond_to?(:instance_id) ? false : true
+    end
+    
+    def extract(cat)
+      miqVm.extract(cat)
+    end
+    
+    def unmount
+      @miqVm.unmount unless @miqVm.nil?
+      unmap_volumes
+    end
+  
+    def ebs_ids
+      ids = []
+      if ebs_instance?
+        @ec2_obj.block_device_mappings.each do |dev|
+          ids << dev.ebs.volume_id
+        end
+      else
+        @ec2_obj.block_device_mappings.each do |dev|
+          ids << dev.ebs.snapshot_id
+        end
+      end
+      ids
+    end
+  
+    def zone_name
+      aim = InstanceMetadata.new
+      aim.metadata('placement/availability-zone')
+    end
+    
+    def miqVm
```

Comment:
@hsong-rh can you be more specific?  The extract method in which class / repo?  I still can't find it.